jax/test_util.py

```diff
@@ -373,6 +373,31 @@ def rand(shape, dtype):
 
   return rand
 
+def rand_some_nan():
+  """Return a random sampler that produces nans in floating types."""
+  rng = npr.RandomState(1)
+  base_rand = rand_default()
+
+  def rand(shape, dtype):
+    """The random sampler function."""
+    if not onp.issubdtype(dtype, onp.floating):
```

Comment:
It looks like there's a bug both in this code and in the `rand_some_inf_and_nan` function below from which it was cloned for complex dtypes. Since complex types aren't subtypes of `onp.floating`, this case will apply for complex numbers, meaning we won't get any nans (or infs in  the other function) for complex  numbers. I  think you can fix it by swapping the order of  the two `if` blocks?